packages/components/Menu/src/Menu/useMenu.ts

```diff
@@ -61,6 +54,13 @@ const useMenuOpenState = (
 
   const state = isControlled ? open : openInternal;
 
+  // Due to how events get fired we get double notifications
+  // for the same event causing us to immediately reopen
+  // a menu when we close it. Adding in a delay to prevent
+  // this behavior.
+  const delayOpen = 150;
```

Comment:
This can stay as a global constant. This doesn't need to be stored separately per component instance.